v3/gssapi/client.go

```diff
@@ -212,3 +216,48 @@ func (client *Client) NegotiateSaslAuth(input []byte, authzid string) ([]byte, e
 
 	return output, nil
 }
+
+func UnmarshalWrapToken(wt *gssapi.WrapToken, data []byte, expectFromAcceptor bool) error {
+	// Check if we can read a whole header
+	if len(data) < 16 {
+		return errors.New("bytes shorter than header length")
```

Comment:
If we anticipate any kind of code handling of these errors it may be kinder to actually create named errors (`var ErrInvalidGSSAPIFooBarHeaderLength = "bytes shorter ..."`), that can be used with the `errors.Is` and related helper functions.